### PR TITLE
Sum of digits dart

### LINKED ISSUE #5415: Sum of digits in dart
## 💥 Proposal

To calculate sum of digits - dart program

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Ayushjain2205**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Code for sum of digits using dart added

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #5415

### CODE REVIEW

File: Dart/maths/SumOfDigits.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    number ~/= 10;
+    sum += digit;
+  }
+  return sum;
+}
+
+// Main Function, with driver code
+void main() {
+  print("Enter a number:");
+  int input = int.parse(stdin.readLineSync()!);
```
Comment (atarax665) [inline Dart/maths/SumOfDigits.dart:19]:
```suggestion
  int input = int.parse(stdin.readLineSync());
```

File: Dart/maths/SumOfDigits.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import 'dart:io';
+
+// Calculate factorial of number
```
Comment (atarax665) [inline Dart/maths/SumOfDigits.dart]: Factorial?

File: Dart/maths/SumOfDigits.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  }
+  return sum;
+}
+
+// Main Function, with driver code
+void main() {
+  print("Enter a number:");
+  int input = int.parse(stdin.readLineSync()!);
+
+  // Call factorial function
```
Comment (atarax665) [inline Dart/maths/SumOfDigits.dart]: Factorial?

File: Dart/maths/SumOfDigits.dart [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  int sum_of_digits = sum_digits(input);
+  print("Sum of digits in $input is $sum_of_digits");
+}
+/**
+Enter a number:
+401
+Sum of digits in 401 is 50
```
Comment (atarax665) [inline Dart/maths/SumOfDigits.dart]:
```suggestion
Sum of digits in 401 is 5
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Dart/maths/SumOfDigits.dart [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+void main() {
+  print("Enter a number:");
+  int input = int.parse(stdin.readLineSync()!);
+
+  // Call sum of digits function
+  int sum_of_digits = sum_digits(input);
+  print("Sum of digits in $input is $sum_of_digits");
+}
+/**
+Enter a number:
```
Comment (bhav09) [inline Dart/maths/SumOfDigits.dart:29]: YOu need to add the time complexity of the code before sample I/O. Rest seems fine.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Minor Changes.

**Comment by Ayushjain2205** [comment]
@bhav09 Changes made

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**